### ISSUE TITLE
FileNotFoundException: No entry for content://media/external/images/media/23925

### ISSUE THREAD

**Issue by misaochan** [nicolas-raoul/apps-android-commons#134]
Sent by Mainudin Samro on 16 Apr:

```
USER_COMMENT=
ANDROID_VERSION=4.2.2
APP_VERSION_NAME=1.12
BRAND=Intex_Cloud_Y1
PHONE_MODEL=Intex Cloud Y1
CUSTOM_DATA=
STACK_TRACE=java.lang.RuntimeException: java.io.FileNotFoundException: No entry for content://media/external/images/media/23925
at fr.free.nrw.commons.upload.UploadService.uploadContribution(UploadService.java:178)
at fr.free.nrw.commons.upload.UploadService.handle(UploadService.java:114)
at fr.free.nrw.commons.upload.UploadService.handle(UploadService.java:27)
at fr.free.nrw.commons.HandlerService$ServiceHandler.handleMessage(HandlerService.java:19)
at android.os.Handler.dispatchMessage(Handler.java:107)
at android.os.Looper.loop(Looper.java:194)
at android.os.HandlerThread.run(HandlerThread.java:60)
Caused by: java.io.FileNotFoundException: No entry for content://media/external/images/media/23925
at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:146)
at android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:617)
at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:747)
at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:644)
at android.content.ContentResolver.openInputStream(ContentResolver.java:479)
at fr.free.nrw.commons.upload.UploadService.uploadContribution(UploadService.java:176)
... 6 more
java.io.FileNotFoundException: No entry for content://media/external/images/media/23925
at android.database.DatabaseUtils.readExceptionWithFileNotFoundExceptionFromParcel(DatabaseUtils.java:146)
at android.content.ContentProviderProxy.openTypedAssetFile(ContentProviderNative.java:617)
at android.content.ContentResolver.openTypedAssetFileDescriptor(ContentResolver.java:747)
at android.content.ContentResolver.openAssetFileDescriptor(ContentResolver.java:644)
at android.content.ContentResolver.openInputStream(ContentResolver.java:479)
at fr.free.nrw.commons.upload.UploadService.uploadContribution(UploadService.java:176)
at fr.free.nrw.commons.upload.UploadService.handle(UploadService.java:114)
at fr.free.nrw.commons.upload.UploadService.handle(UploadService.java:27)
at fr.free.nrw.commons.HandlerService$ServiceHandler.handleMessage(HandlerService.java:19)
at android.os.Handler.dispatchMessage(Handler.java:107)
at android.os.Looper.loop(Looper.java:194)
at android.os.HandlerThread.run(HandlerThread.java:60)
```

**Comment by misaochan** [comment]
Well, this is expected I guess since the relevant code is:

```
catch(FileNotFoundException e) {
            throw new RuntimeException(e);
        }
```

But what do we want to replace it with?

**Comment by misaochan** [comment]
Attempted fix at #137
